src/NServiceBus.Testing.Fakes/TestableAuditContext.cs

```diff
@@ -2,37 +2,55 @@
 {
     using System;
     using System.Collections.Generic;
+    using NServiceBus.Audit;
     using Pipeline;
     using Transport;
 
     /// <summary>
     /// A testable implementation of <see cref="IAuditContext" />.
     /// </summary>
-    public partial class TestableAuditContext : TestableBehaviorContext, IAuditContext
+    public partial class TestableAuditContext : TestableBehaviorContext, IAuditContext, IAuditActionContext
     {
         /// <summary>
-        /// Contains the information added by <see cref="AddedAuditData" />.
+        /// Address of the audit queue.
         /// </summary>
-        public IDictionary<string, string> AddedAuditData { get; } = new Dictionary<string, string>();
+        public string AuditAddress { get; set; } = "audit queue address";
 
         /// <summary>
-        /// Address of the audit queue.
+        /// The configured time to be received for audit messages.
         /// </summary>
-        public string AuditAddress { get; set; } = "audit queue address";
+        public TimeSpan? TimeToBeReceived { get; } = null;
 
         /// <summary>
         /// The message to be audited.
         /// </summary>
         public OutgoingMessage Message { get; set; } = new OutgoingMessage(Guid.NewGuid().ToString(), new Dictionary<string, string>(), new byte[0]);
 
+        /// <summary>
+        /// Metadata of the audited message.
+        /// </summary>
+        public Dictionary<string, string> AuditMetadata { get; set; } = new Dictionary<string, string>();
+
+        /// <summary>
+        /// Gets the messages, if any, this audit operation should result in.
+        /// </summary>
+        public AuditAction AuditAction { get; set; }
+
+        IReadOnlyDictionary<string, string> IAuditActionContext.AuditMetadata => AuditMetadata;
+
         /// <summary>
         /// Adds information about the current message that should be audited.
         /// </summary>
         /// <param name="key">The audit key.</param>
         /// <param name="value">The value.</param>
         public void AddAuditData(string key, string value)
         {
-            AddedAuditData.Add(key, value);
+            AuditMetadata.Add(key, value);
+        }
+
+        public IAuditActionContext PreventChanges()
```

Comment:
TestableRecoverabilityContext does

```
       public IRecoverabilityActionContext PreventChanges()
        {
            IsLocked = true;
            return this;
        }

        /// <summary>
        /// True if the recoverability action was locked.
        /// </summary>
        public bool IsLocked { get; set; } = false;
```

should we align? And while we are at it should we make the `IsLocked` get only?

Reply:
Agreed, I've aligned and also made the recoverability one get only